cabot/metricsapp/forms/elastic.py

```diff
@@ -22,3 +24,44 @@ def clean_urls(self):
             return input_urls
         except ConnectionError:
             raise ValidationError('Invalid Elasticsearch host url(s).')
+
+
+class ElasticsearchStatusCheckForm(StatusCheckForm):
+    class Meta:
+        model = ElasticsearchStatusCheck
+        fields = (
+            'name',
+            'source',
+            'queries',
+            'check_type',
+            'warning_value',
+            'high_alert_importance',
+            'high_alert_value',
+            'time_range',
+            'frequency',
+            'active',
+            'retries',
+        )
+
+        def __init__(self, *args, **kwargs):
+            ret = super(ElasticsearchStatusCheckForm, self).__init__(*args, **kwargs)
+            self.fields['source'].queryset = ElasticsearchSource.objects.all()
+            return ret
+
+    def clean_queries(self):
+        """
+        Make sure input queries are formatted correctly.
+        """
+        queries = self.cleaned_data['queries']
+        try:
+            query_list = json.loads(queries)
+        except ValueError:
+            raise ValidationError('Queries must be json-parsable')
+
+        for query in query_list:
+            try:
+                validate_query(query)
```

Comment:
Agreed that it shouldn't be here, but I'm thinking about moving it to model.clean() instead of model.save().

When you submit data in a form, if you raise a ValidationError in model.clean(), it'll be caught and you can show an error message on the form page. If an error is raised in model.save(), you'll go to a django error page (or just get the message "an unhandled exception occurred," which is way worse ux. 

If we make a Cabot API and we're creating/updating models not through forms, I think we'd want to call [full_clean](https://docs.djangoproject.com/en/1.11/ref/models/instances/#django.db.models.Model.full_clean)() on all models before saving to make sure there aren't any other issues with the input data (it also cleans fields and validates unique constraints, which I think we'd want). As long as we do that when we're saving we'll still be validating queries, so I think validation in model.clean() would work out for a cabot API.

Lmk if you see any issues with this!